app/services/parameter_safe_render.rb

```diff
@@ -0,0 +1,38 @@
+# Class to parse ERB with or without Safemode rendering. Needs a set
+# of variables, usually something like:
+#   @allowed_vars = { :host => @host }
+# so that <%= @host.name %> has the right @host variable
+#
+class ParameterSafeRender
+  include Foreman::Renderer
+  def initialize(host)
+    @host = host
+  end
+
+  def render(value)
+    render_object(value)
+  end
+
+  private
+
+  def render_object(object)
+    return object if (!Setting[:interpolate_erb_in_parameters])
+
+    # recurse over object types until we're dealing with a String
+    case object
+    when String
+      render_string object
+    when Array
+      object.map { |v| render_object v }
+    when Hash
+      object.merge(object) { |k, v| render_object v }
+    else
+      # Don't know how to parse this, send it back
+      object
+    end
+  end
+
+  def render_string(string)
+    render_safe(string, ::Foreman::Renderer::ALLOWED_HOST_HELPERS, :host => @host)
```

Comment:
This does not allow access to other, non-host related macros, is there a reason to that? I can imagine 
access to `snippet` macro, `foreman_server_url` macro and possibly others would be useful. So perhaps using ALLOWED_HELPERS would be better?

Reply:
The main reason was https://github.com/theforeman/foreman/blob/398cb5b2d27e71ca14c4aebbbc5acbd16baa9a59/lib/foreman/renderer.rb#L218, where for non-host related methods, we use some tweaking based on the request params, and it would not work as expected. I can include it there though.